### PR TITLE
fix: get pipeline by name not id in Data Factory and Synapse

### PR DESCRIPTION
**Pull request by yehiaelbehery**
Use get_pipeline_by_name in the case of Data Factory or Synapse as they use the name not ID similar to Fabric.

### CODE REVIEW

**Comment by arjendev** [comment]
Hi @yehiaelbehery,

Could you explain why the current implementation is not working as expected? For Data Factory we set the pipeline_name as the `pipeline_id` in the `deserializer`, because names are unique in Data Factory and thus we can treat them as a `pipeline_id`.

Curious to hear your motivation!

**Comment by yehiaelbehery** [comment]
Hello @arjendev,

Mainly because there was a method called get_pipeline_by_name and the error message using "name". So I assumed this is the design.
If it's not the design, I would still opt-in to use the separate function to document and express this difference between data factory and fabric implementation, which may impact other functionality.

File: src/data_factory_testing_framework/_test_framework.py [hunk 1]
```diff
@@ -146,10 +157,17 @@ def evaluate_activities(self, activities: List[Activity], state: PipelineRunStat
...
-                        )
+                        pipeline: Pipeline = None
+                        if self.framework_type == TestFrameworkType.Fabric:
+                            pipeline = self.get_pipeline_by_id(
+                                # Note: in the future they will probably rename this to referenceId for Fabric
+                                execute_pipeline_activity.type_properties["pipeline"]["referenceName"],
+                            )
+                        else:
+                            pipeline = self.get_pipeline_by_name(
+                                # Note: in the future they will probably rename this to referenceId for Fabric
```
Comment (arjendev) [inline src/data_factory_testing_framework/_test_framework.py:168]: You can remove this comment here as it's not applicable for the other TestFrameworkTypes

**Review by arjendev — CHANGES_REQUESTED** [review]
Fair enough @yehiaelbehery,

Two very minor comments and then we can merge this as well!

File: src/data_factory_testing_framework/_test_framework.py [hunk 2]
```diff
@@ -54,21 +54,32 @@ def __init__(
...
             else:
                 self._repository = DataFactoryRepository([])
-        elif framework_type == TestFrameworkType.Synapse:
+        elif self._framework_type == TestFrameworkType.Synapse:
             raise NotImplementedError("Synapse test framework is not implemented yet.")
 
         self._should_evaluate_child_pipelines = should_evaluate_child_pipelines
 
+    @property
+    def framework_type(self) -> str:
```
[the remark this answers is not in the document]
Comment (yehiaelbehery) [reply src/data_factory_testing_framework/_test_framework.py]: Done.

File: src/data_factory_testing_framework/_test_framework.py (hunk 1 above)
Comment (yehiaelbehery) [reply src/data_factory_testing_framework/_test_framework.py]: Done.

**Review by arjendev — APPROVED** [review]
Thanks once more for your useful contributions!

**Pull request merged**